crates/task-impls/src/vid.rs

```diff
@@ -71,10 +72,25 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     *view_number,
                 )
                 .await;
+                let payload_commitment = vid_disperse.payload_commitment;
+                let shares = VidDisperseShare::from_vid_disperse(vid_disperse.clone());
+                let mut consensus = self.consensus.write().await;
+                for share in shares {
+                    let s = share.clone();
```

Comment:
Instead of needing to copy each share, would we just use `into_iter` over `shares` so we can consume it? If I'm understanding correctly, it looks like iterating over shares returns `&share`, but we need to own it to call `to_proposal`, but I *think* if we use `shares.into_iter()` we can obtain a `share`. This might be an oversimplification since I haven't run it.

Reply:
`share` is already owned here and `shares` is being consumed. the `let s = share.clone()` is to get around some type inference issue with the rust  compiler.  For some reason the compiler does not like `let key: <TYPES as NodeType>::SignatureKey = share.recipient_key.clone();`.